lib/galaxy/datatypes/converters/CONVERTER_uncompressed_to_gz.xml

```diff
@@ -0,0 +1,70 @@
+<tool id="CONVERTER_uncompressed_to_gz" name="Compress file(s) using gzip" version="0.1.0">
+    <requirements>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        #if $files.type=="individual":
+            gzip -c ${files.input} > output_file.gz
+            #set $type = "${files.input.extension}.gz"
+        #else:
+            mkdir outdir &&
+            #for $i,$file in enumerate($files.input):
+                gzip -c ${file} > outdir/${file.element_identifier}.gz &&
+            #end for
+            ls outdir
+        #end if -->
+
+    ]]></command>
+    <inputs>
+        <conditional name="files">
+            <param name="type" type="select" label="Compress individual file or a collection?">
+                <option value="individual">Individual file</option>
+                <option value="collection">Collection</option>
+            </param>
+            <when value="individual">
+                <param name="input" type="data" format="data" label="Select data to compress"/>
+            </when>
+            <when value="collection">
+                    <param name="input" type="data_collection" collection_type="list" label="Select data to compress"/>
+            </when>
+        </conditional>
+    </inputs>
+    <outputs>
+        <data name="output_file" format="gz" from_work_dir="output_file.gz" label="GZIP compression on ${on_string}">
```

Comment:
Might be good to make this specific for file types that have compressed variants, eg fastq.gz. Probably only possible with galaxy.json.

Reply:
We were wanting to make this as generic as possible for use in exporting later.

Reply:
Please take this extra ste, otherwise this isn't useful as a converter and could just be a standard tool. For export purposes we have #13149